### PR TITLE
Create registration and user profile endpoint

### PR DESCRIPTION
**Pull request by Unique-01**
Fixes part of #4 
This includes registration for student account only, email verification, password change, password reset, and profile

### CODE REVIEW

**Comment by Unique-01** [comment]
> Fixes part of #4 This includes registration for student account only, email verification, password change, password reset, and profile

Added new endpoints for content creator registration and also admin approval for the account

**Comment by leehomeecky** [comment]
pleas stop trying to merge PR's that hasn't been reviewed.

File: backend/accounts/templates/email_verification.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html>
+<body>
+    <p>Hello {{ user.username }},</p>
+    
+    <p>Please verify your email by clicking on the following link:</p>
+    <p><a href="{{ verification_url }}">{{ verification_url }}</a></p>
```
Comment (leehomeecky) [inline backend/accounts/templates/email_verification.html:7]: change `<p><a href="{{ verification_url }}">{{ verification_url }}</a></p>` to `<p><a href="{{ verification_url }}">verify</a></p>`

or something better

File: backend/accounts/views.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        token = serializer.validated_data['token']
+        user_id = serializer.validated_data['user_id']
+
+        try:
+            uid = force_str(urlsafe_base64_decode(user_id))
+            user = User.objects.get(pk=uid)
+        except (TypeError, ValueError, OverflowError, User.DoesNotExist):
+            return Response({'detail': 'Invalid token.'}, status=status.HTTP_400_BAD_REQUEST)
```
Comment (leehomeecky) [inline backend/accounts/views.py:145]: the status code should be 403, un authorized

File: backend/accounts/views.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
```
Comment (leehomeecky) [inline backend/accounts/views.py:183]: status code should be 403, un authorized

File: backend/accounts/views.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        if not default_token_generator.check_token(user, token):
+            return Response({'detail': 'Invalid token.'}, status=status.HTTP_400_BAD_REQUEST)
```
Comment (leehomeecky) [inline backend/accounts/views.py:185]: try refactor this, and let the error come once

**Review by Odamola6 — CHANGES_REQUESTED** [review]

File: backend/accounts/views.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+            if not default_token_generator.check_token(user, token):
+                raise ValueError
+        except (TypeError, ValueError, OverflowError, User.DoesNotExist):
+            return Response({'detail': 'Invalid token.'}, status=status.HTTP_401_UNAUTHORIZED)
+
+        user.is_active = True
+        user.user_type = 'content_creator'
+        user.save()
+        self.send_activated_account_email(user)
+        return Response({'detail': 'Account activated successful.'})
```
[the remark this answers is not in the document]
Comment (Odamola6) [reply backend/accounts/views.py:198]: This hasn't been changed yet

**Review by Odamola6 — CHANGES_REQUESTED** [review]

File: backend/accounts/views.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Unique-01) [reply backend/accounts/views.py:198]: That is just the response sent to the front end

**Review by Odamola6 — APPROVED** [review]

**Pull request merged**